zk/watcher.go

```diff
@@ -102,6 +104,10 @@ func (w *Watcher) reconnect() error {
 	}
 	w.conn = conn
 
+	// QUESTION: why do need to set connection timeout here?
```

Comment:
If I remember correctly the server disconnects were unreliable, so we started keeping track on the client side.

Reply:
The sample code implies that zk.Connect() is a synchronous call. When we reach here, we have already connected to zookeeper.

Reply:
I debugged it a little. The first event we get is is an event with zk.StateConnecting. We get it immediately. It doesn't seem we have to wait the timeout here.